tests/test_density_RDF.py

```diff
@@ -74,44 +72,44 @@ def test_invalid_rdf(self):
         with pytest.raises(ValueError):
             freud.density.RDF(r_max=1, bins=10, r_min=-1)
 
-    def test_random_point(self):
+    @pytest.mark.parametrize("r_min", [0, 0.1, 3.0])
+    def test_random_point(self, r_min):
         r_max = 10.0
         bins = 10
         num_points = 10000
         tolerance = 0.1
         box_size = r_max * 3.1
 
-        for r_min in (0, 0.1, 3.0):
-            box, points = freud.data.make_random_system(box_size, num_points)
-            # This test is slow, and since it's a validation of the underlying
-            # algorithm and not the API we don't need to test all possible
-            # inputs, so we only test the fastest one (AABBQuery).
-            nq = freud.locality.AABBQuery(box, points)
-            neighbors = {"mode": "ball", "r_max": r_max, "exclude_ii": True}
-
-            rdf = freud.density.RDF(bins, r_max, r_min)
-
-            if r_min != 3.0:
-                rdf.compute(nq, neighbors=neighbors, reset=False)
-            else:
-                rdf.compute(nq, neighbors=neighbors)
-
-            assert rdf.box == box
-            correct = np.ones(bins, dtype=np.float32)
-            npt.assert_allclose(rdf.rdf, correct, atol=tolerance)
-
-            # Numerical integration to compute the running coordination
-            # number will be highly inaccurate, so we can only test up to
-            # a limited precision. Also, since dealing with nonzero r_min
-            # values requires extrapolation, we only test when r_min=0.
-            ndens = points.shape[0] / box.volume
-            dr = (r_max - r_min) / bins
-            bin_boundaries = np.array(
-                [r_min + dr * i for i in range(bins + 1) if r_min + dr * i <= r_max]
-            )
-            bin_volumes = 4 / 3 * np.pi * np.diff(bin_boundaries**3)
-            avg_counts = rdf.rdf * ndens * bin_volumes
-            npt.assert_allclose(rdf.n_r, np.cumsum(avg_counts), rtol=tolerance)
+        box, points = freud.data.make_random_system(box_size, num_points)
+        # This test is slow, and since it's a validation of the underlying
+        # algorithm and not the API we don't need to test all possible
+        # inputs, so we only test the fastest one (AABBQuery).
+        nq = freud.locality.AABBQuery(box, points)
+        neighbors = {"mode": "ball", "r_max": r_max, "exclude_ii": True}
+
+        rdf = freud.density.RDF(bins, r_max, r_min)
+
+        if r_min != 3.0:
```

Comment:
Why do we set `reset=False` explicitly for values besides 3.0? This is a special case of the parameterization but we don't have a comment explaining why this special case is used. This needs a comment -- or perhaps separated into another test, if it's covering something a case of the RDF behavior that is sufficiently unique.

Reply:
I believe the reason for `reset` being set to `False` explicitly is just to test that `compute` will work when `reset` is called as an argument, and the 3.0 value was chosen arbitrarily. I added a comment to explain this.

Reply:
Testing `reset` behavior should be done in a separate test function.